test/extended/builds/pipeline.go

```diff
@@ -13,36 +16,98 @@ import (
 	"github.com/openshift/origin/test/extended/util/jenkins"
 )
 
+func debugAnyJenkinsFailure(br *exutil.BuildResult, name string, oc *exutil.CLI, dumpMaster bool) {
+	if !br.BuildSuccess {
+		fmt.Fprintf(g.GinkgoWriter, "\n\n START debugAnyJenkinsFailure\n\n")
+		j := jenkins.NewRef(oc)
+		jobLog, err := j.GetLastJobConsoleLogs(name)
+		if err == nil {
+			fmt.Fprintf(g.GinkgoWriter, "\n %s job log:\n%s", name, jobLog)
+		} else {
+			fmt.Fprintf(g.GinkgoWriter, "\n error getting %s job log: %#v", name, err)
+		}
+		if dumpMaster {
+			exutil.DumpDeploymentLogs("jenkins", oc)
+		}
+		fmt.Fprintf(g.GinkgoWriter, "\n\n END debugAnyJenkinsFailure\n\n")
+	}
+}
+
 var _ = g.Describe("[builds][Slow] openshift pipeline build", func() {
 	defer g.GinkgoRecover()
 	var (
-		jenkinsTemplatePath       = exutil.FixturePath("..", "..", "examples", "jenkins", "jenkins-ephemeral-template.json")
-		mavenSlavePipelinePath    = exutil.FixturePath("..", "..", "examples", "jenkins", "pipeline", "maven-pipeline.yaml")
-		orchestrationPipelinePath = exutil.FixturePath("..", "..", "examples", "jenkins", "pipeline", "mapsapp-pipeline.yaml")
-		blueGreenPipelinePath     = exutil.FixturePath("..", "..", "examples", "jenkins", "pipeline", "bluegreen-pipeline.yaml")
+		jenkinsTemplatePath    = exutil.FixturePath("..", "..", "examples", "jenkins", "jenkins-ephemeral-template.json")
+		mavenSlavePipelinePath = exutil.FixturePath("..", "..", "examples", "jenkins", "pipeline", "maven-pipeline.yaml")
+		//orchestrationPipelinePath = exutil.FixturePath("..", "..", "examples", "jenkins", "pipeline", "mapsapp-pipeline.yaml")
+		blueGreenPipelinePath = exutil.FixturePath("..", "..", "examples", "jenkins", "pipeline", "bluegreen-pipeline.yaml")
+
+		oc                       = exutil.NewCLI("jenkins-pipeline", exutil.KubeConfigPath())
+		ticker                   *time.Ticker
+		j                        *jenkins.JenkinsRef
+		dcLogFollow              *exec.Cmd
+		dcLogStdOut, dcLogStdErr *bytes.Buffer
+		setup                    = func() {
+			// Deploy Jenkins
+			g.By(fmt.Sprintf("calling oc new-app -f %q", jenkinsTemplatePath))
+			err := oc.Run("new-app").Args("-f", jenkinsTemplatePath).Execute()
+			o.Expect(err).NotTo(o.HaveOccurred())
+
+			g.By("waiting for jenkins deployment")
+			err = exutil.WaitForADeploymentToComplete(oc.KubeClient().Core().ReplicationControllers(oc.Namespace()), "jenkins", oc)
+			o.Expect(err).NotTo(o.HaveOccurred())
+
+			j = jenkins.NewRef(oc)
+
+			g.By("wait for jenkins to come up")
+			_, err = j.WaitForContent("", 200, 10*time.Minute, "")
+
+			if err != nil {
+				exutil.DumpDeploymentLogs("jenkins", oc)
+			}
+
+			o.Expect(err).NotTo(o.HaveOccurred())
+
+			if os.Getenv(jenkins.DisableJenkinsMemoryStats) == "" {
```

Comment:
Correct and Correct.  And in case it wasn't clear, we already merged in the use of this env var.  I simply moved it from a private var in `jenkins_plugin.go` to a public var in `monitor.go` since it is being leveraged in different places now.